Module2-Evaluations.js

```diff
@@ -141,242 +140,117 @@ function createMonthSheetAndOverallColumn() {
-  Logger.log('Creating pool of potential evaluators.');
-  const potentialEvaluators = [...allAmbassadorsEmails, ...allAmbassadorsEmails, ...allAmbassadorsEmails];
-
-  // Dictionary to track how many times an evaluator is assigned
-  const ambassadorCount = {};
+/**
+ * Attempts to assign evaluators to submitters in a single attempt.
+ * Filters out invalid or empty data during the process.
+ * @param {Array} validSubmitters - Array of valid submitters.
+ * @param {Array} allEvaluators - Array of eligible evaluators.
+ * @returns {Object} - Contains assignments and count of "Has No Evaluator".
+ */
+function attemptSingleAssignment(validSubmitters, allEvaluators) {
+  Logger.log('Starting single attempt to assign evaluators.');
 
-  // Initialize assignments
   const assignments = [];
+  const evaluatorPool = [...allEvaluators, ...allEvaluators, ...allEvaluators]; // Triplicate evaluator pool for flexibility
+  const evaluatorCounts = {}; // Track evaluator usage
   let countHasNoEvaluator = 0;
 
-  // Assign evaluators to each submitter
-  Logger.log('Assigning evaluators to submitters.');
-  submittersEmails.forEach((submitter) => {
+  validSubmitters.forEach((submitter) => {
     const reviewers = [];
     for (let i = 0; i < 3; i++) {
-      const availableEvaluators = potentialEvaluators.filter(
+      const availableEvaluators = evaluatorPool.filter(
         (evaluator) =>
-          evaluator !== submitter && // exclude submitter evaluating themselves
-          (ambassadorCount[evaluator] || 0) < 3 && // no more than 3 assignments per evaluator
-          !reviewers.includes(evaluator) // no duplicate evaluators for same submitter
+          evaluator !== submitter && // Exclude self-evaluation
+          (evaluatorCounts[evaluator] || 0) < 3 && // Limit evaluators to 3 assignments
+          !reviewers.includes(evaluator) // Avoid duplicate evaluators for the same submitter
       );
 
       if (availableEvaluators.length === 0) {
         reviewers.push('Has No Evaluator');
-        Logger.log(`No available evaluators for ${submitter} in round ${i + 1}`);
         countHasNoEvaluator++;
+        Logger.log(`No available evaluators for submitter ${submitter} in round ${i + 1}.`);
         continue;
       }
 
       const randomIndex = Math.floor(Math.random() * availableEvaluators.length);
       const selectedEvaluator = availableEvaluators[randomIndex];
 
       reviewers.push(selectedEvaluator);
-      ambassadorCount[selectedEvaluator] = (ambassadorCount[selectedEvaluator] || 0) + 1;
-      Logger.log(`Assigned ${selectedEvaluator} to ${submitter} in round ${i + 1}`);
+      evaluatorCounts[selectedEvaluator] = (evaluatorCounts[selectedEvaluator] || 0) + 1;
+      Logger.log(`Assigned evaluator ${selectedEvaluator} to submitter ${submitter} in round ${i + 1}.`);
     }
     assignments.push({ submitter, reviewers });
   });
 
-  // Determine which evaluators were never used
-  Logger.log('Determining unassigned evaluators.');
```

Comment:
This can come back in once the loop is discarded.
I don't see where ele you are notifying unassigned evaluators? We want to do that.